app/src/main/java/com/example/antitheft/structure/GPSCommandHandler.java

```diff
@@ -0,0 +1,66 @@
+package com.example.antitheft.structure;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.example.antitheft.MainActivity;
+
+/**
+ * This class' scope is used to send a command, or to manage incoming commands, replying
+ * to the one who sent the command.
+ *
+ * @author Francesco Bau'
+ * @version 0.1
+ * @since 25/02/2020
+ */
+public class GPSCommandHandler {
+
+    //**********SENDING**********
+
+    /**
+     * This method sends a given message (command).
+     *
+     * @param smsMessage The message to send. It can't be null.
+     * @see com.eis.smslibrary.SMSMessage
+     * @see com.eis.smslibrary.SMSManager
+     */
+    public void sendMessage(@NonNull SMSMessage smsMessage) {
+        SMSManager.getInstance().sendMessage(smsMessage);
+    }
+
+    //**********RECEPTION AND REPLY**********
+
+    /**
+     * This method handles the received command, from {@link GPSCommandReceiver#onMessageReceived(SMSMessage)},
+     * replying with the last known GPS location, retrieved by {@link MainActivity#getCurrentLocation()}.
+     *
+     * @param command The received command. It can't be null.
+     * @see com.eis.smslibrary.SMSMessage
+     * @see GPSCommandReceiver#onMessageReceived(SMSMessage)
+     * @see MainActivity#getCurrentLocation()
+     */
+    protected void onCommandReceived(@NonNull SMSMessage command) {
```

Comment:
You are totally right! I hadn't thought about that...

Though that brings me to change a lot of things, by just changing it to SMSPeer...
I hope I won't forget something behind...

Reply:
Hold on.
Actually, I need that to be called "`onCommandReceived()`", and I need a `SMSMessage` instance, because, in the future, there's not gonna be just the LOCATE command, but other commands as well, so, after `GPSCommandReceived` noticed that's a valid command, it leaves the rest to `GPSCommandHandler`.

But, you made me think I should better call this method `onLocateCommandReceived(@ NonNull SMSPeer peer)` instead, and, in the future, I'm gonna have more methods, e.g. `onRingCommandReceived(@ NonNull SMSPeer peer)`.

So basically, there are these 2 choices as well:
1) leaving the method as it is, and use a `switch-case` to check which kind of command is requested.
    - PRO: more reliable, if well-implemented, lighter GPSCommandReceiver.
    - CON: more complex, possible redundancy (GPSCommandReceiver already checks if it's a command)
2) rename it `onLocateCommandReceived(@ NonNull SMSPeer peer)` ,  and, in the future, add more similar methods.
    - PRO: easier method(s)
    - CON: GPSCommandReceiver needs to filter better the incoming messages, and the method is more fragile, if someone on purpose calls that method, even if it's not a command => double-check is needed.

I think (2) might be the best choice, but the issue is that `GPSCommandReceiver` needs to filter better.
What do you think about it? One of these two, or the one you suggested?

Reply:
In my opinion, you could keep in the GPSCommandReceiver the onMessageReceived method which analyzes the command received and invokes the right class.
In this GPSCommandHandler class you rename this method as onLocateCommandReceived. If in the future you want to add the ring option then you will do a RingCommandHandler and add there the method.

Then GPSCommandReceiver analyzes the command and invokes the right method either of GPSCommandHandler or of RingCommandHandler.

Reply:
Ok, so you agree with option (2), too, right?

I'll change that when I can, then!

Thank you for your opinion! ✌️

Reply:
Wait, what do you mean by "RingCommandHandler"?

You mean like proposal (2), so method `onRingCommandReceived(SMSPeer)` inside GPSCommandHandler (I guess I should find a better name, in the future...), or you mean another class?

Reply:
Another class (e.g. `RingCommandHandler`) with method `onRingCommandReceived(SMSPeer)`.
Different managers/handlers for different scopes.
It is advice for keeping classes and uses separate.

Reply:
Oh, I see....
Thank you for the advice! ✌️ 
I will surely think about that!